src/monodroid/jni/embedded-assemblies-zip.cc

```diff
@@ -0,0 +1,449 @@
+	uint16_t              compression_method;
+	uint32_t              local_header_offset;
+	uint32_t              data_offset;
+	uint32_t              file_size;
+	char                 *entry_name;
+	char                 *file_name;
+
+	ssize_t nread = read (fd, buf.get (), cd_size);
+	if (static_cast<size_t>(nread) != cd_size) {
+		log_fatal (LOG_DEFAULT, "Failed to read Central Directory from the APK archive %s. %s (nread: %d; errno: %d)", apk_name, std::strerror (errno), nread, errno);
+		exit (FATAL_EXIT_NO_ASSEMBLIES);
+	}
+
+	for (size_t i = 0; i < cd_entries; i++) {
+		bool result = zip_read_entry_info (buf.get (), cd_size, buf_offset, compression_method, local_header_offset, file_size, entry_name);
+		simple_pointer_guard<char> entry_name_guard = entry_name;
+		file_name = entry_name_guard.get ();
+
+#ifdef DEBUG
+		log_warn (LOG_DEFAULT, "%s entry: %s", apk_name, file_name);
+#endif
+		if (!result) {
+			log_fatal (LOG_DEFAULT, "Failed to read Central Directory info for entry %u in APK file %s", i, apk_name);
+			exit (FATAL_EXIT_NO_ASSEMBLIES);
+		}
+
+		if (!zip_adjust_data_offset (fd, local_header_offset, data_offset)) {
+			log_fatal (LOG_DEFAULT, "Failed to adjust data start offset for entry %u in APK file %s", i, apk_name);
+			exit (FATAL_EXIT_NO_ASSEMBLIES);
+		}
+#ifdef DEBUG
+		log_warn (LOG_DEFAULT, "    ZIP: local header offset: %u; data offset: %u; file size: %u", local_header_offset, data_offset, file_size);
+#endif
+		if (compression_method != 0)
+			continue;
+
+#if defined (DEBUG)
+		if (utils.ends_with (file_name, ".jm")) {
+			md_mmap_info map_info   = md_mmap_apk_file (fd, data_offset, file_size, file_name, apk_name);
+			add_type_mapping (&java_to_managed_maps, apk_name, file_name, (const char*)map_info.area);
+			continue;
+		}
+		if (utils.ends_with (file_name, ".mj")) {
+			md_mmap_info map_info   = md_mmap_apk_file (fd, data_offset, file_size, file_name, apk_name);
+			add_type_mapping (&managed_to_java_maps, apk_name, file_name, (const char*)map_info.area);
+			continue;
+		}
+#endif
+		if (strncmp (prefix, file_name, prefix_len) != 0)
+			continue;
+
+		// assemblies must be 4-byte aligned, or Bad Things happen
+		if ((data_offset & 0x3) != 0) {
+			log_fatal (LOG_ASSEMBLY, "Assembly '%s' is located at bad offset %lu within the .apk\n", file_name, data_offset);
+			log_fatal (LOG_ASSEMBLY, "You MUST run `zipalign` on %s\n", strrchr (apk_name, '/') + 1);
+			exit (FATAL_EXIT_MISSING_ZIPALIGN);
+		}
+
+		bool entry_is_overridden = !should_register (strrchr (file_name, '/') + 1);
+
+		if ((utils.ends_with (file_name, ".mdb") || utils.ends_with (file_name, ".pdb")) &&
+				register_debug_symbols &&
+				!entry_is_overridden &&
+				bundled_assemblies != nullptr) {
+			md_mmap_info map_info = md_mmap_apk_file(fd, data_offset, file_size, file_name, apk_name);
+			if (register_debug_symbols_for_assembly (file_name, (bundled_assemblies) [bundled_assemblies_count - 1],
+						(const mono_byte*)map_info.area,
```

Comment:
Indentation here doesn't match context; this line should be *un*-indented one tab stop, to align with the following line.  Or the following line should be indented a tab stop to align with this line.